server/app/controllers/applicant/RedirectController.java

```diff
@@ -107,32 +111,35 @@ public CompletionStage<Result> programBySlug(Http.Request request, String progra
                         // for this program, redirect to program version associated
                         // with that application if so.
                         if (programForExistingApplication.isPresent()) {
+                          long programId = programForExistingApplication.get().id();
                           return CompletableFuture.completedFuture(
-                              redirect(
-                                      controllers.applicant.routes.ApplicantProgramReviewController
-                                          .review(
-                                              applicantId,
-                                              programForExistingApplication.get().id()))
-                                  .flashing("redirected-from-program-slug", programSlug));
+                              goToActiveProgram(programId, applicantId, programSlug, request));
+                        } else {
+                          return programService
+                              .getActiveProgramDefinitionAsync(programSlug)
```

Comment:
Can you add handling for the case where there's no active program found? We should still clear the session in that case

Reply:
We clear the session in both cases, calling `redirectToReviewPage` (previously `goToActiveProgram`), which has the `removingFromSession` line.

Reply:
getActiveProgramDefinitionAsync() can throw a ProgramNotFoundException. I'm referring to handling that exception.

Reply:
Added. It doesn't seem possible to modify the `Request` directly, only to call `removingFromSession` on a new `Result`. So we have to return a `notFound()` (a Result), and the pass the exception's message. But since it's buried two layers deep, we get this:

<img width="1092" alt="image" src="https://github.com/civiform/civiform/assets/30369272/583134f2-eef3-42a3-bb87-2b861e1f38c0">

We can call `ex.getCause().getCause()` to get the underlying message, but that couples the code with the cause  being two layers deep, so the simplest thing to do might be to leave the exception as-is. Open to your thoughts if I missed something.

Reply:
I think that's fine, just want to make sure they're not stuck in a loop.